mesonbuild/dependencies/boost.py

```diff
@@ -594,7 +594,8 @@ def filter_libraries(self, libs: T.List[BoostLibraryFile], lib_vers: str) -> T.L
         # mlog.debug('    - vscrt: {}'.format(vscrt))
         libs = [x for x in libs if x.static == self.static or not self.explicit_static]
         libs = [x for x in libs if x.mt == self.multithreading]
-        libs = [x for x in libs if x.version_matches(lib_vers)]
+        if not mesonlib.is_openbsd():
```

Comment:
specifically this will always be the build machine.

To get at the host machine you need to use `self.env.machines[self.for_machine]` to get the correct machine information